common/src/main/java/io/netty/util/AsciiString.java

```diff
@@ -714,6 +714,34 @@ public int indexOf(CharSequence subString, int start) {
     }
 
     /**
+     * Searches in this string for the index of the specified char {@code ch}.
+     * The search for the char starts at the specified offset {@code start} and moves towards the end of this string.
+     *
+     * @param ch the char to find.
+     * @param start the starting offset.
+     * @return the index of the first occurrence of the specified char {@code ch} in this string,
+     * -1 if found no occurrence.
+     */
+    public int indexOf(char ch, int start) {
+        if (start < 0) {
```

Comment:
@Scottmitch Sorry for confused previous message, of course you are right and this methods are safe in String. I revisited code, but I think this check for negative index should leave as it is, because in my method this value using in another checks before calling forEachByte. forEachByte now accept length parameter, which is calculated using something like ```length() - startIndex```, so startIndex must be normalized before calling forEachByte with current signature. I think it's better to leave it as it is with just a strong check.